rasa/core/policies/ted_policy.py

```diff
@@ -634,46 +635,89 @@ def run_training(
                 These may or may not be used by the function depending
                 on how the policy is trained.
         """
-        if not self.finetune_mode:
-            # This means the model wasn't loaded from a
-            # previously trained model and hence needs
-            # to be instantiated.
-            self.model = self.model_class()(
-                model_data.get_signature(),
-                self.config,
-                isinstance(self.featurizer, MaxHistoryTrackerFeaturizer),
-                self._label_data,
-                self._entity_tag_specs,
+        if(self.config[USE_GPU]):
```

Comment:
Can this be refactored so there isn't duplication of the training calls? E.g. using a context manager

Reply:
There are two ways do de-duplicate the code here:
1. We could put the code into a function
```
if condition:
  # don't use context manager
   run_code_that_needs_to_run_in_both_cases() 
else:
    with special_context_manager:  # i.e. cpu device placement
       run_code_that_needs_to_run_in_both_cases()
``` 
2. A nice way that avoids having to create another function is to use a [nullcontext](https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext). With such a `nullcontext`, we can write ...
```
with (special_context_manager if condition else nullcontext):
   run_code_run_with_or_without_context_manager() 
``` 
   -  ... to achieve the same as in 1.
   - (Btw, `nullcontext`s are available for Python >= 3.7 and [luckily we dropped support for Python <= 3.6](https://github.com/RasaHQ/rasa/blob/main/pyproject.toml#L66) :)